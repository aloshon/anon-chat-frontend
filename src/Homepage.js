import React, {useContext, useEffect} from "react";
import UserContext from "./UserContext";
import GroupChatCard from "./GroupChatCard";
import Container from "react-bootstrap/Container";
import {fetchInvitedGroupChats} from "./actionCreators";
import {useSelector, useDispatch, shallowEqual} from "react-redux";
import {useHistory} from "react-router-dom";

function Homepage(){
    const {user} = useContext(UserContext);
    console.log(user)


    /** 
     * isInviting allow users to click on group chats they created
     * then the page will load the invite component with the 
     * group chat title on top asking who you want to add
     * The guest list should be provided so from there
     * check is guest is already invited or if the guest
     * list exeeds 10 people
     */
    /**  isInviting is an object with 3 keys and values, currently is 
     * set to false but once true it will return a different component.
     * unique_id is the group chat's unique_id to check if user is the creator of group chat,
     * group_chat_id is the group_chat_id that the user is going to add
     * guests into. Title is the title of the group chat
     * 
    */
    const groupChatTitles = useSelector(state => state.titles, shallowEqual);
    const dispatch = useDispatch();
    const history = useHistory();

    useEffect(() => {
        dispatch(fetchInvitedGroupChats(user.id))
    }, [dispatch, user.id]);

    function displayInvitePage(unique_id){
        return history.push(`/invite/${unique_id}`)
    }


    if(groupChatTitles.length === 0) return <h1>No chats yet!</h1>;
    console.log(groupChatTitles);
    if(groupChatTitles){
        return (
            <Container style={{marginBottom: "15px"}}>
                {/* {message.length > 0 && 
                <div className="alert alert-success">
                    {message}
                </div>} */}
                {groupChatTitles.map(gc => (
                    <GroupChatCard 
                    key={gc.unique_id}
                    unique_id={gc.unique_id}
                    title={gc.title}
                    description={gc.description}
                    timestamp={gc.timestamp}
                    creator={gc.creator_id}
                    displayInvitePage={displayInvitePage}
                    />
                ))}
            </Container>
        )
    }

    return <h1>Loading...</h1>
}

export default Homepage;