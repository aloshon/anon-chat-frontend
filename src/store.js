import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);