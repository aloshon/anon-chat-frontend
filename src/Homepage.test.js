import { render, screen } from '@testing-library/react';
import Homepage from './Homepage';
import { Provider } from "react-redux";
import { store } from "./store"
import {BrowserRouter} from 'react-router-dom';
import UserContext from "./UserContext";


describe("homepage should render", () => {
    const user = {
        id: 1,
        username: "user1",
        isAdmin: false,
        blockList: []
    }

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <UserContext.Provider value={{user}}>
                <Homepage/>
            </UserContext.Provider>
        </BrowserRouter>
      </Provider>
    );
    test("renders homepage, no group chats though", () => {
      const noChatsText = screen.getByText("No chats yet!");
      expect(noChatsText).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });