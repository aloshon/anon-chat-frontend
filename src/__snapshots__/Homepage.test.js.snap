// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage should render matches snapshot 1`] = `<DocumentFragment />`;
